HIP/hip-551.md

```diff
@@ -74,54 +78,89 @@ Inner transactions, by design:
     */
-    repeated bytes bodyBytes = 1;
+    Key batchKey = 63;
 
-    /**
-     * The signatures on the body with the new format, to authorize the transactions
-     */
-    SignatureMap sigMap = 2;
+    oneof data {
+      // Existing values omitted.
+
+      AtomicBatchTransactionBody atomicBatchTransactionBody = 64;
+    }
 }
+
 ```
 
-## Backwards Compatibility
+The AtomicBatchTransactionBody will contain the transaction(s) that make up the batch.
 
-## Security Implications
+```protobuf
+message AtomicBatchTransactionBody {
+    repeated Transaction transactions = 1;
+}
+
+```
 
-As part of ensuring the integrity of batched transactions, certain security measures are emphasized:
+The top-level `AtomicBatchTransactionBody` must be signed by the `batchKey` specified in the included transactions. There may only be one `batchKey` set for all the transactions in the batch. i.e. Inner transaction X cannot have a different `batchKey` than inner transaction Y.
 
-1. Inner transactions within a batch cannot be reordered post-batch-signature.
-2. All signatures on the batch pertain solely to the batch and not the inner transactions.
+We must return what transaction failed in the batch when the batch fails.
 
-## How to Teach This
+```protobuf
+message TransactionReceipt {
+    // Existing properties omitted.
+    
+    repeated ResponseCodeEnum batchTransactionStatus = 16;
+}
+```
 
-## Reference Implementation
+### Security Enhancements
 
-## Rejected Ideas
-We rejected the idea to add the support of conditional branching to the batch transactions,
-because we don't think it's useful compared to the increase of complications in implementing the HIP.
+1. The batch transaction system prevents malicious actors from tampering with transactions by requiring them to be processed as a single unit.
+2. A `batchKey` ensures only the trusted assembler (e.g., Alice) can create a valid batch. Nodes attempting to disassemble the batch or alter it will invalidate the transaction.
 
-## Open Issues
+### Batch Transaction Timeout
 
-## References
+All inner transactions must execute within the standard `transactionValidStart` and `transactionValidDuration` (typically 3 minutes) to ensure they remain valid during batch processing.
 
-## Copyright/license
+## Backward Compatibility
+
+This proposal is backward compatible with existing Hedera transaction models, as it introduces a new message type for batch transactions without changing current transaction structures.
+
+## Security Implications
+
+Security is reinforced by ensuring that inner transactions cannot be executed outside the batch and by preventing malicious nodes from modifying batch transactions. The use of a `batchKey` ensures that only the trusted entity can assemble and submit the batch.
+
+## Rejected Ideas
+
+Conditional branching within batch transactions was rejected as it would introduce unnecessary complexity without significant value for the proposed use cases.
```

Comment:
I am not sure what conditional branching means. Please define that term.

Reply:
Conditional branching in transaction chains determines the next transaction to execute (A or B) based on whether the previous transaction succeeded or failed.
Tx1 --Success--> TxA
Tx1 --Failure--> TxB

Reply:
Added to the hip for clarification. Thanks @SimiHunjan !